### ISSUE TITLE
UI darkmode / lightmode bug

### ISSUE THREAD

**Issue by kevinkrs** [TablePlus/TablePlus#1782]
1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
Its an UI bug.

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
3.1.2. Build number 296, the latest version

3. The steps to reproduce this issue:
The UI doesn't change back to lightmode automatically. I have my Mac setup in a way that on day MacOS will use lightmode but at the evening it will automaticly use darkmode.

However, when the Mac switches from darkmode to lightmode the app does not change back to lightmode as well. Not even after a restart of TablePlus.

The only way to get it back to lightmode is by changing the setting in MacOS settings back to lightmode and then select automatically  again.

Perhaps there is an fix for this?

**Comment by huyphams** [comment]
I've used auto theme for months but couldn't reproduce the issue, it always works on my laptop. That's weird. I will keep an eye on it.

**Comment by kevinkrs** [comment]
<img width="1303" alt="Schermafbeelding 2020-03-23 om 10 59 45" src="https://user-images.githubusercontent.com/8309250/77310839-519e2380-6cff-11ea-9e51-22aa04ff0ae9.png">

I forgot to attach a screenshot. As you can see my setting is at automatic, all applications such as finder are in lightmode except TablePlus.

1 More note: I used my Macbook Pro on an external (4k) monitor with scaling on. After i was done i closed the lid and shut down my laptop. Perhaps you can replicate this a bit better with the given information.

**Comment by huyphams** [comment]
Here is mine. let me try to shutdown the laptop. I've never shutdown my laptop. Just close the lid.
<img width="1483" alt="abc" src="https://user-images.githubusercontent.com/3654622/77311165-31d42300-6d32-11ea-8826-72b95482b2cd.png">

**Comment by kevinkrs** [comment]
Im curious what result it will give you. I'm not completly sure if the same result will happen if you just close the lid. But its important that you shut off the computer at night (dark mode) and turn it on at sunrise (light mode).

I will try it tonight with just letting my laptop sleep.

**Comment by huyphams** [comment]
> I'm curious what result it will give you.
It will match 100% with the macOS theme. light-light, dark-dark.

**Comment by kevinkrs** [comment]
Hmm, so i tried it today, with just my laptop in standby without turning it off. It seems TablePlus is refreshing the UI back to lightmode. I dont know if its because the new MacOS 10.15.4 update.

**Comment by romanzipp** [comment]
I've also noticed that the TablePlus scheme is not toggling with the macOS appearance (darkmode everytime, no matter at which time I opened or closed the application).

I've just switched the TablePlus theme from Auto to Light and backwards, now it works just fine. Probably corrupted default settings?

**Comment by romanzipp** [comment]
Whoops, just noticed that the dark theme is still active, if the computer has been powered down with darkmode enabled and started in daytime.
temporarily. 
Mac mini @ macOS 10.15.4

**Comment by kevinkrs** [comment]
> Whoops, just noticed that the dark theme is still active, if the computer has been powered down with darkmode enabled and started in daytime.
> 
> Mac mini @ macOS 10.15.4
> TablePlus 3.3.0

Not completly, if my macbook turns off with automatic darkmode enabled, and it stays off untill daylight and i boot it back up again in the morning when i'm at the office MacOS instantly switches back to lightmode but TablePlus stays in darkmode untill i manually change the UI in MacOS from automatic to lightmode and back to automatic again. Changing de UI from automatic to light back to automatic in TablePlus does not help.

Got the issue yesterday again :(

**Comment by huyphams** [comment]
I got it, so the bug happens after the mac restart. Let me check it.

**Comment by kevinkrs** [comment]
Exactly, it seems that TablePlus doesn't check if the theme has changed by startup.

**Comment by romanzipp** [comment]
The same problem also occurred in an IDEA darkmode sync plugin. Maybe this issue can help resolving the bug: https://github.com/gilday/dark-mode-sync-plugin/issues/7

**Comment by huyphams** [comment]
Hi guys, please try this patch and see how it goes: https://www.dropbox.com/s/hijhy8j8b6ilpdr/TablePlus.dmg?dl=0
Thank you!

**Comment by kevinkrs** [comment]
Thanks, I installed this version. There is no way to change the timing of DarkMode in the settings of MacOS so we can test in day too?

**Comment by kevinkrs** [comment]
Goodmorning, the patch does not seem to resolve the issue, i replicated the steps by turning the laptop of at night when automatic darkmode is enabled and turning it on in the morning. Same result
![Uploading Schermafbeelding 2020-04-21 om 09.55.08.png…]()

**Comment by romanzipp** [comment]
Can confirm, bug still exists

![Screen Shot 2020-04-22 at 08 48 54](https://user-images.githubusercontent.com/11266773/79949611-1a45a280-8476-11ea-9794-41b8b4647656.png)

**Comment by hsul4n** [comment]
Any updates!!

**Comment by datishans** [comment]
Bug exists on  3.10.0 (348), running macOS Big Sur, beta 10

**Comment by xpol** [comment]
And when change theme it will trigger TablePlus menu.

https://youtu.be/sS6TLzi8-2o

**Comment by marcheimendinger** [comment]
Still the same issue here running TablePlus 3.12.2 and macOS 11.1.

It's the only app having this issue on my installation.

**Comment by philippgerard** [comment]
Same problem here with 11.2 and build 358 of TP.

**Comment by porcherie** [comment]
same problem BigSur 11.2.3, TablePlus Version 3.12.5 (364) (MacBook m1)

**Comment by alexbensimon** [comment]
Same problem with TablePlus 3.12.6 and MacOS 11.2.3.

**Comment by Rykian** [comment]
Still no news?

Not a big thing, but it's the only holding me to buy the app. If a small bug can't be fixed what does it says about app support?

I don't know any other app having the same problem.

**Comment by leandroruel** [comment]
im facing same issue, but not in MAC in my linux mint.

**Comment by huyphams** [comment]
Hi, do you guy still seeing this bug in the newest release?

**Comment by kevinkrs** [comment]
I have not seen it anymore on my M1 Macbook Pro for a while, I dont know if that makes an difference or not compared to Intel based systems.

**Comment by xpol** [comment]
Does the code here help?
https://developer.apple.com/forums/thread/118974
